backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java

```diff
@@ -138,28 +138,22 @@ public Receiver createReceiver() {
         return RabbitFlux.createReceiver(new ReceiverOptions().connectionMono(connectionMono));
     }
 
-    public Mono<Connection> getConnectionMono() {
-        return connectionMono;
-    }
-
     @Override
     public Mono<? extends Channel> getChannelMono() {
-        return Mono.fromCallable(this::borrow);
+        return borrow();
     }
 
-    private Channel borrow() {
-        Channel channel = tryBorrowFromPool();
-        borrowedChannels.add(channel);
-        return channel;
-    }
-
-    private Channel tryBorrowFromPool() {
-        return Mono.fromCallable(this::borrowFromPool)
+    private Mono<Channel> borrow() {
+        return tryBorrowFromPool()
             .doOnError(throwable -> LOGGER.warn("Cannot borrow channel", throwable))
             .retryBackoff(MAX_BORROW_RETRIES, MIN_BORROW_DELAY, FOREVER, Schedulers.elastic())
```

Comment:
are we retrying whatever the exception?

Reply:
Agreed, but IMO it's an other concern. Can we keep it for another PR?

Reply:
Is it a question? I mean, you merged the PR before I can answer

Reply:
Sorry if I went a bit fast here.

The comment was unrelated to the change.

Also Duc is doing some testing related to this here: https://github.com/linagora/james-project/pull/3282?notification_referrer_id=MDE4Ok5vdGlmaWNhdGlvblRocmVhZDgxNjI0MDU4NDo2OTI4NzQw&notifications_query=is%3Adone